checkpoint.py

```diff
@@ -133,6 +137,22 @@ def path_tuple_to_string(path: tuple) -> str:
     return "/".join(pieces)
 
 
+"""
+For get_load_path_str(),
+introducing a simple caching mechanism to avoid recomputing regex matches for paths that have already been processed.
+"""
+@lru_cache(maxsize=None)
```

Comment:
Can we be sure that not limiting the size within the lru cache won't cause problems?

Reply:
Absolutely, setting maxsize=None was optimistic on memory availability. However, I see your point about the risks of letting the cache grow indefinitely. We shall consider use cases to choose a sensible max size that makes sense.

Reply:
From reading the docs, and given modern cache sizes, 32MB may be a sensible default.

Reply:
Cool, I'll update it.